tests/test_sync.py

```diff
@@ -437,6 +437,64 @@ def test_sync_dry_run_would_uninstall(echo, from_line):
     echo.assert_has_calls(expected_calls, any_order=True)
 
 
+@mock.patch("piptools.sync.click.confirm")
+@mock.patch("piptools.sync.click.echo")
+def test_sync_ask_declined_would_install(echo, confirm, from_line):
+    """
+    Sync with --ask option does a dry run if the user declines
+    """
+    confirm.return_value = False
+
+    to_install = {from_line("django==1.8"), from_line("click==4.0")}
+
+    sync(to_install, set(), ask=True)
+
+    expected_calls = [
+        mock.call("Would install:"),
+        mock.call("  django==1.8"),
+        mock.call("  click==4.0"),
+    ]
+    echo.assert_has_calls(expected_calls, any_order=True)
+
+    confirm.assert_called_once_with("Would you like to proceed with these changes?")
```

Comment:
Could you add here an assert to make sure that sync() won't install/uninstall anything, since the side-effect declines confirmation?
```python
check_call.assert_not_called()
```

Reply:
Done!